core/drivers/crypto/hisilicon/include/sec_main.h

```diff
@@ -0,0 +1,376 @@
+#define SEC_CKEY_OFFSET_V3	13
+#define SEC_CALG_OFFSET_V3	4
+#define SEC_AKEY_OFFSET_V3	9
+#define SEC_MAC_OFFSET_V3	4
+#define SEC_AUTH_ALG_OFFSET_V3	15
+#define SEC_CIPHER_AUTH_V3	0xbf
+#define SEC_AUTH_CIPHER_V3	0x40
+#define SEC_AI_GEN_OFFSET_V3	2
+#define SEC_SEQ_OFFSET_V3	6
+#define SEC_ICV_LEN_OFFSET_V3	4
+#define SEC_DK_LEN_OFFSET_V3	16
+#define SEC_KEY_SEL_OFFSET_V3	21
+
+struct hisi_sec_sqe_type2 {
+	/*
+	 * mac_len: 0~4 bits
+	 * a_key_len: 5~10 bits
+	 * a_alg: 11~16 bits
+	 */
+	uint32_t mac_key_alg;
+
+	/*
+	 * c_icv_len: 0~5 bits
+	 * c_width: 6~8 bits
+	 * c_key_len: 9~11 bits
+	 * c_mode: 12~15 bits
+	 */
+	uint16_t icvw_kmode;
+
+	/* c_alg: 0~3 bits */
+	uint8_t c_alg;
+
+	uint8_t rsvd4;
+	/*
+	 * a_len: 0~23 bits
+	 * iv_offset_l: 24~31 bits
+	 */
+	uint32_t alen_ivllen;
+
+	/*
+	 * c_len: 0~23 bits
+	 * iv_offset_h: 24~31 bits
+	 */
+	uint32_t clen_ivhlen;
+
+	uint16_t auth_src_offset;
+	uint16_t cipher_src_offset;
+	uint16_t cs_ip_header_offset;
+	uint16_t cs_udp_header_offset;
+	uint16_t pass_word_len;
+	uint16_t dk_len;
+	uint8_t salt3;
+	uint8_t salt2;
+	uint8_t salt1;
+	uint8_t salt0;
+
+	uint16_t tag;
+	uint16_t rsvd5;
+
+	/*
+	 * c_pad_type: 0~3 bits
+	 * c_pad_len: 4~11 bits
+	 * c_pad_data_type: 12~15 bits
+	 */
+	uint16_t cph_pad;
+	/* c_pad_len_field: 0~1 bits */
+	uint16_t c_pad_len_field;
+
+	uint64_t long_a_data_len;
+	uint64_t a_ivin_addr;
+	uint64_t a_key_addr;
+	uint64_t mac_addr;
+	uint64_t c_ivin_addr;
+	uint64_t c_key_addr;
+	uint64_t data_src_addr;
+	uint64_t data_dst_addr;
+
+	/*
+	 * done: 0 bit
+	 * icv: 1~3 bits
+	 * csc: 4~6 bits
+	 * flag: 7~10 bits
+	 */
+	uint16_t done_flag;
+
+	uint8_t error_type;
+	uint8_t warning_type;
+	uint8_t mac_i3;
+	uint8_t mac_i2;
+	uint8_t mac_i1;
+	uint8_t mac_i0;
+	uint16_t check_sum_i;
+	uint8_t tls_pad_len_i;
+	uint8_t rsvd12;
+	uint32_t counter;
+};
+
+struct hisi_sec_sqe {
+	/*
+	 * type:  0~3 bits;
+	 * cipher: 4~5 bits;
+	 * auth: 6~7 bits;
+	 */
+	uint8_t type_auth_cipher;
+	/*
+	 * seq: 0 bits;
+	 * de: 1~2 bits;
+	 * scene: 3~6 bits;
+	 * src_addr_type: 7 bits;
+	 */
+	uint8_t sds_sa_type;
+	/*
+	 * src_addr_type: 0~1 bits not used now.
+	 * dst_addr_type: 2~4 bits;
+	 * mac_addr_type: 5~7 bits;
+	 */
+	uint8_t sdm_addr_type;
+
+	uint8_t rsvd0;
+	/*
+	 * nonce_len(type): 0~3 bits;
+	 * huk: 4 bit;
+	 * key_s: 5 bit
+	 * ci_gen: 6~7 bits
+	 */
+	uint8_t huk_ci_key;
+	/*
+	 * ai_gen: 0~1 bits;
+	 * a_pad : 2~3 bits;
+	 * c_s : 4~5 bits;
+	 */
+	uint8_t ai_apd_cs;
+	/*
+	 * rhf(type2): 0 bit;
+	 * c_key_type: 1~2 bits;
+	 * a_key_type: 3~4 bits
+	 * write_frame_len(type2): 5~7bits;
+	 */
+	uint8_t rca_key_frm;
+
+	uint8_t iv_tls_ld;
+	/* the other scenario */
+	struct hisi_sec_sqe_type2 type2;
+};
+
+struct bd3_stream_scene {
+	uint64_t c_ivin_addr;
+	uint64_t long_a_data_len;
+
+	/*
+	 * auth_pad: 0~1 bits
+	 * stream_protocol: 2~4 bits
+	 * reserved: 5~7 bits
+	 */
+	uint8_t auth_pad;
+	uint8_t plaintext_type;
+	uint16_t pad_len_1p3;
+} __packed __aligned(4);
```

Comment:
Why is this `__packed __aligned(4)`?

Reply:
`struct hisi_sec_bd3_sqe` isn't used anywhere. Can you wait with it until it's used so we can see how it's going to be used?